pages/[username]/[collectionId]/[nftId]/index.tsx

```diff
@@ -4,26 +4,42 @@ import { GetServerSideProps } from 'next';
 import GalleryRedirect from 'scenes/_Router/GalleryRedirect';
 import { MetaTagProps } from 'pages/_app';
 import { openGraphMetaTags } from 'utils/openGraphMetaTags';
+import { Nft } from 'types/Nft';
+import { getImageSrcSet } from 'utils/imageSrcSet';
 
 type NftDetailPageProps = MetaTagProps & {
-  nftId?: string;
+  nft?: Nft;
 };
 
-export default function NftDetailPage({ nftId }: NftDetailPageProps) {
-  if (!nftId) {
+export default function NftDetailPage({ nft }: NftDetailPageProps) {
+  if (!nft) {
     // Something went horribly wrong
     return <GalleryRedirect to="/" />;
   }
 
-  return <GalleryRoute element={<NftDetailPageScene nftId={nftId} />} footerIsFixed />;
+  return <GalleryRoute element={<NftDetailPageScene nft={nft} />} footerIsFixed />;
 }
 
 export const getServerSideProps: GetServerSideProps<NftDetailPageProps> = async ({ params }) => {
   const username = params?.username ? (params.username as string) : undefined;
   const nftId = params?.nftId ? (params.nftId as string) : undefined;
+
+  // retrieve nft from db
+  const res = await fetch(`${process.env.NEXT_PUBLIC_API_BASE_URL}/glry/v1/nfts/get?id=${nftId}`);
+  const data = await res.json();
+
+  if (!data?.nft) {
+    return {
+      notFound: true,
+    };
+  }
+
+  // generate srcset for image resizing
+  const imageSrcSet = getImageSrcSet(data.nft.image_original_url, 600);
+
```

Comment:
general Q: would we normally place this fetch in `getServerSideProps`? is that the correct way to handle in nextjs?

Reply:
yup according to the docs! https://nextjs.org/docs/basic-features/data-fetching/get-server-side-props#fetching-data-on-the-client-side

leaving here for others